### PR TITLE
Filtered tests are not skipped any more.

### PR DESCRIPTION
**Pull request by rinslow**

### CODE REVIEW

File: src/rotest/core/utils/pretty.py [hunk 1]
```diff
@@ -347,7 +347,9 @@ def get_columns():
     if sys.platform in ("linux", "linux2", "darwin"):
         # Both OS X (darwin) and Linux support the 'stty' command
         try:
-            _, columns = os.popen("stty size 2>/dev/null").read().split()
+            with os.popen("stty size 2>/dev/null") as pipe:
```
Comment (UnDarkle) [inline src/rotest/core/utils/pretty.py]: on windows?

File: src/rotest/core/filter.py [hunk 2]
```diff
@@ -29,25 +29,36 @@ def get_tags(test):
...
 
-    test._tags = tags
-    return tags
+    if isinstance(test, type) and issubclass(test, (TestCase, TestSuite)):
+        return test.TAGS[:] + [test.__name__]
+
+    if isinstance(test, (TestBlock, TestFlow)):
+        return test.TAGS[:] + [test.get_name()]
+
+    if isinstance(test, type) and issubclass(test, (TestBlock, TestFlow)):
```
Comment (UnDarkle) [inline src/rotest/core/filter.py]: dude, we don't filter test blocks =\

File: src/rotest/cli/client.py [hunk 3]
```diff
@@ -178,9 +180,15 @@ def main(*tests):
     if len(tests) == 0:
         tests = discover_tests_under_paths(options.paths)
 
+    # Filter tests by tags
+    tests = [test for test in tests if match_tags(get_tags(test),
+                                                  supplied_filter
+                                                  if supplied_filter
+                                                  else "*")]
```
Comment (UnDarkle) [inline src/rotest/cli/client.py]: else "*"?!
put this filtering in 'if' instead

File: src/rotest/cli/client.py [hunk 4]
```diff
@@ -178,9 +180,15 @@ def main(*tests):
     if len(tests) == 0:
         tests = discover_tests_under_paths(options.paths)
 
+    # Filter tests by tags
+    tests = [test for test in tests if match_tags(get_tags(test),
```
Comment (UnDarkle) [inline src/rotest/cli/client.py]: does it work?
i think the 'get_tags' function is supposed to get instances and not classes...

File: src/rotest/core/filter.py [hunk 5]
```diff
@@ -29,25 +29,36 @@ def get_tags(test):
...
     Args:
-        test (TestSuite / TestCase): test item instance.
+        test (TestSuite / TestCase / TestFlow / TestBlock): test item instance.
 
     Returns:
         list. tags of the test item.
     """
-    if test._tags is not None:
-        return test._tags
+    if isinstance(test, (TestCase, TestSuite)):
```
Comment (UnDarkle) [inline src/rotest/core/filter.py]: why and what are all those changes to this file?! =\

File: src/rotest/core/utils/pretty.py (hunk 1 above)
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: It's in an if-clause that checks for the current platform

**Pull request closed without merging**